raiden/blockchain_events_handler.py

```diff
@@ -447,7 +447,9 @@ def handle_channel_batch_unlock(raiden: "RaidenService", event: Event):
                 sender=partner,
             )
             if state_change_record is not None:
-                canonical_identifier = state_change_record.data.balance_proof.canonical_identifier
+                canonical_identifier = (
+                    state_change_record.data.balance_proof.canonical_identifier  # type: ignore
```

Comment:
I thought about introducing a new marker type with `balance_proof` in it. However this PR was big enough already.